page/layouts/shortcodes/imghash.html

```diff
@@ -0,0 +1,35 @@
+{{/*
+  *    Shortcode: imghash (page/layouts/shortcodes/imghash.html)
+  *
+  *    Description:
+  *    Add image with hash parameter
+  *
+  *    Example Shortcode Usage: 
+  *    {{% imghash src="/pathto/image" %}}
+  * 
+  *    Optional:
+  *    {{% imghash src="/pathto/img" info="Alt Text" %}}
+  *    {{% imghash src="/pathto/img" info="Alt Text" param="classes=shadow" %}}
+  *
+*/}}
+
+{{ $getPath     := .Get "src" | default false }}
+{{ $getFilePath := slicestr $getPath 3 }}
+
+{{ $getInfo   := .Get "info" | default " " }}
+{{ $getParam  := .Get "param" }}
+
+{{ if os.FileExists $getFilePath }}
+
+ {{ $getFileHash := md5 (os.Stat $getFilePath).ModTime.Unix }}
```

Comment:
Not sure if there is any advantage, but Hugo offers a [Fingerprint](https://gohugo.io/hugo-pipes/fingerprint/) method.

Reply:
As I understand it, [fingerprint](https://gohugo.io/hugo-pipes/fingerprint/) is used in conjunction with resource, which changes the filename itself. But i really do not know?

Reply:
Which would be fine - or what are your objections against it?

Reply:
I have tried it but have not been able to implement it. Somehow I got the fingerprint "xyz.123456789.png". But this does not work, because the file itself does not exist than. I think this works only in combination with hugo static resources within the static/assets folder. But help is welcome!

Reply:
Tested this one:

```
{{ $imgPath:= "composer-update.png" }}
{{ $imgPathResArr := .Page.Resources.GetMatch $imgPath }}
{{ $imgPathRes :=  $imgPathResArr | resources.Fingerprint  }}
<img src="{{$imgPathRes.RelPermalink}}">
```
The hugo build folder contains then two files each:
```
composer-update.png
composer-update.7ad89b74dfd1a89bbf59e64cfeb3d3e25eee8e69350a844459273399bbfcd610.png
```

This will only work if the image is in the same folder (next to the .md file)?
Can't manage to address any image as a resource here?